synapse/storage/databases/main/registration.py

```diff
@@ -1343,18 +1343,22 @@ async def create_registration_token(
         Returns:
             Whether the row was inserted or not.
         """
-        return await self.db_pool.simple_insert(
-            "registration_tokens",
-            values={
-                "token": token,
-                "uses_allowed": uses_allowed,
-                "pending": 0,
-                "completed": 0,
-                "expiry_time": expiry_time,
-            },
-            or_ignore=True,
-            desc="create_registration_token",
-        )
+        try:
+            await self.db_pool.simple_insert(
+                "registration_tokens",
+                values={
+                    "token": token,
+                    "uses_allowed": uses_allowed,
+                    "pending": 0,
+                    "completed": 0,
+                    "expiry_time": expiry_time,
+                },
+                desc="create_registration_token",
+            )
+        except self.database_engine.module.IntegrityError:
```

Comment:
Note that raising integrity errors will still appear in postgres' logs, so it's best to avoid them where we can. See https://github.com/matrix-org/synapse/pull/10442.

While it would cost another database query, it may be a good idea to check for existing tokens with a `simple_select` first instead.

Reply:
Done in 01a74da6f7b42a1b146210c51aa0928602d4e85f